docs(gitlab-ci): adding gitlab-ci documentation (#479)
nuxt-hub/core: docs/content/docs/1.getting-started/3.deploy.md

- Separate applications should be deployed using different workflow jobs.

## Gitlab CI

This section will guide you to implement the GitLab CI.
The integration with GitLab CI builds on the [Usage with CI/CD](https://hub.nuxt.com/docs/getting-started/deploy#usage-with-cicd) section,
make sure you have those two variables `NUXT_HUB_PROJECT_KEY` and `NUXT_HUB_USER_TOKEN`

### Setup Gitlab CI Variables

::note

> It is bad practice to place the user token inside the repository.
> Therefore, we place it in the GitLab CI Variables

Reviewer: ```suggestion
It is good practice to place the user token inside GitLab CI Variables.
```

::

1. Open your Repository (on GitLab) > Settings > CI/CD > Variables
2. Click on _Add variable_
3. Set Visibility to Masked and hidden
4. Remove protected Flag (since we may use this variable also on non-protected branches)
5. Give a Key name = `NUXT_HUB_USER_TOKEN`
6. Paste your `NUXT_HUB_USER_TOKEN`
7. Click Add variable

::note
The `NUXT_HUB_PROJECT_KEY` is used later in [Configure Projects for Gitlab](#configure-projects-for-gitlab).
::

### For Monorepos with multiple apps (Optional Step)

::note

> if you do not use a monorepo with multiple apps, jump to [Configure Projects for Gitlab](#configure-projects-for-gitlab)

Reviewer: ```suggestion
If you do not use a monorepo with multiple apps, jump to [Configure Projects for Gitlab](#configure-projects-for-gitlab).
```
